Description: use the user settings, and not the default system ones, so we
 get the user's background when the screen is locked.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: no, is a Ubuntu-specific preference

Index: gnome-screensaver-3.2.0/src/gs-manager.c
===================================================================
--- gnome-screensaver-3.2.0.orig/src/gs-manager.c	2012-02-09 21:23:41.000000000 -0500
+++ gnome-screensaver-3.2.0/src/gs-manager.c	2012-02-09 21:25:51.884447527 -0500
@@ -555,13 +555,8 @@
                                      gint       n_keys,
                                      GSManager   *manager)
 {
-#if 0
-        /* FIXME: since we bind user settings instead of system ones,
-         *        watching for changes is no longer valid.
-         */
         gnome_bg_load_from_preferences (manager->priv->bg,
                                         manager->priv->settings);
-#endif
 
         return FALSE;
 }
@@ -570,24 +565,9 @@
 get_system_settings (void)
 {
         GSettings *settings;
-        gchar **keys;
-        gchar **k;
 
-        /* FIXME: we need to bind system settings instead of user but
-         *        that's currently impossible, not implemented yet.
-         *        Hence, reset to system default values.
-         */
-        /* TODO: Ideally we would like to bind some other key, screensaver-specific. */
         settings = g_settings_new ("org.gnome.desktop.background");
 
-        g_settings_delay (settings);
-
-        keys = g_settings_list_keys (settings);
-        for (k = keys; *k; k++) {
-                g_settings_reset (settings, *k);
-        }
-        g_strfreev (keys);
-
         return settings;
 }
 
